### PR TITLE
Slew

### PR DESCRIPTION
**Pull request by ellem95**
Adding the ability to calculate satellites as antenna moves across sky

### CODE REVIEW

**Comment by nikojpapa** [comment]
This is based on the `window_finder` branch, right? Can you change the base of the PR to reflect that so we are not reviewing duplicate changes?

File: satellite_determination/path_finder/observation_path_finder.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        self._reservation = reservation
+        self._start_time_str = start_str
+        self._end_time_str = end_str
+
+    def calculate_path(self) -> List[ObservationPath]:
+        observation_path = []
+        observing_location = EarthLocation(lat=str(self._reservation.facility.point_coordinates.latitude), lon=str(self._reservation.facility.point_coordinates.longitude), height=self._reservation.facility.height * u.m)
+        target_coordinates = SkyCoord(self._reservation.facility.right_ascension, self._reservation.facility.declination)
+        #start_time = self._reservation.time.begin
+        #end_time = self._reservation.time.end
```
Comment (nikojpapa) [inline satellite_determination/path_finder/observation_path_finder.py:22]:
```suggestion
```

**Review by nikojpapa — CHANGES_REQUESTED** [review]

File: expected_path.txt
[the remark this answers is not in the document]
Comment (ellem95) [reply expected_path.txt:1]: done

File: satellite_determination/TLEData/test.txt
[the remark this answers is not in the document]
Comment (ellem95) [reply satellite_determination/TLEData/test.txt:1]: removed

File: satellite_determination/custom_dataclasses/observation_path.py
[the remark this answers is not in the document]
Comment (ellem95) [reply satellite_determination/custom_dataclasses/observation_path.py:1]: Deleted unnecessary function

File: satellite_determination/path_finder/observation_path_finder.py (hunk 1 above)
Comment (ellem95) [reply satellite_determination/path_finder/observation_path_finder.py]: done

File: satellite_determination/window_finder.py [hunk 2]
```diff
@@ -72,7 +72,7 @@ def find(self) -> List[SuggestedReservation]:
         ]
 
     def _satellites_overhead(self, reservation: Reservation) -> List[OverheadWindow]:
-        return self._event_finder(list_of_satellites=self._satellites, reservation=reservation).get_overhead_windows()
+        return self._event_finder(list_of_satellites=self._satellites, reservation=reservation, azimuth_altitude_path=None).get_overhead_windows()
```
[the remark this answers is not in the document]
Comment (ellem95) [reply satellite_determination/window_finder.py:75]: WindowFinder is only used if we are doing stationary observations so does not have a ```azimuth_altitude_path```

File: satellite_determination/event_finder/event_finder_rhodesmill/event_finder_rhodesmill.py [hunk 3]
```diff
@@ -1,24 +1,33 @@
...
+        self._path = azimuth_altitude_path
 
     def get_overhead_windows(self):
         ts = load.timescale() #provides time objects with the data tables they need to translate between different time scales: the schedule of UTC leap seconds, and the value of ∆T over time.
-        overhead_windows = []
-        time_start = ts.from_datetime(convert_datetime_to_utc(self._reservation.time.begin)) #changes the reservation datetime to Skyfield Time object
-        time_end = ts.from_datetime(convert_datetime_to_utc(self._reservation.time.end))
+        overhead_windows: list[OverheadWindow] = []
+        time_start = ts.from_datetime(self._reservation.time.begin.replace(tzinfo=pytz.UTC))  # changes the reservation datetime to Skyfield Time object
+        time_end = ts.from_datetime(self._reservation.time.end.replace(tzinfo=pytz.UTC))
```
[the remark this answers is not in the document]
Comment (ellem95) [reply satellite_determination/event_finder/event_finder_rhodesmill/event_finder_rhodesmill.py]: Resolved

[the remark this answers is not in the document]
Comment (ellem95) [reply satellite_determination/event_finder/event_finder_rhodesmill/event_finder_rhodesmill.py]: Messaged in slack. If you look at the code at this file and these line numbers, you'll see its been updated. The other instances are all updated in PR #27 where they were commented on.

**Review by nikojpapa — APPROVED** [review]

**Pull request merged**